### ISSUE TITLE
"class defined inside a class" even when class is a resource (parameterized Class)

### ISSUE THREAD

**Issue by hkraal** [rodjek/puppet-lint#32]
The parameterized class

``` ruby
class foo ($bar = 'default') {
  notify{"The value is: ${bar}": }
}
```

The main class

``` ruby
class bar {
  class { 'foo':
    bar => 'foobar'
  }
}
```

The main class will give the error:

```
WARNING: class defined inside a class on line 2
```

Altough this should be allowed according to the puppet docs at http://docs.puppetlabs.com/guides/parameterized_classes.html

EDIT: I haven't seen this behaviour when it comes to parameterized defines which are called upon within a class. Perhaps the same remedy can be of use here?
